File: packages/typescript/src/permission.ts
export namespace Permission {

	export function isAllowed(permission: Permission, permissions: string[]): boolean {
		return permissions.includes(Admin) || permissions.includes(permission);
	}

	export function getDescription(permission: Permission): string {
		switch (permission) {
		}

		return "No description provided.";
	}
}

export type Permission = ;

File: packages/rust/src/permission.rs
use serde::{Deserialize, Serialize};
use std::borrow::Borrow;

#[derive(Serialize, Deserialize, PartialEq, Eq, Debug)]
pub enum Permission {
}

impl Permission {
	pub fn is_allowed<P: Borrow<Self>>(permission: P, permissions: &[Self]) -> bool {
		permissions.contains(&Permission::Admin) || permissions.contains(permission.borrow())
	}

	pub fn check(&self, permissions: &[Self]) -> bool {
		Self::is_allowed(self, permissions)
	}
}
